internal/gateway/statistics/statistics.go

```diff
@@ -20,22 +21,26 @@ type Counters struct {
 // TODO: concurrent access
 type StatisticsMgr struct {
 	leaseCounters map[string]Counters
+	rwLock        sync.RWMutex
```

Comment:
Also I guess it should be a normal mutex, we are never reading from it. Are we?

Reply:
No we are not, but is it bad? What if we will in the future?

Reply:
No it is not bad, it actually doesn't matter.

But if somebody else is reading the code it will wonder why there is a RWMutex instead of a normal one.

If tomorrow we will need a RW mutex it is much simpler to add, while it is difficult to remove.

But it is not an issue... Just the way I prefer to do stuff...

Reply:
Alright, that makes sense.